File: src/main/java/dev/retrotv/crypt/exception/WrongPaddingException.java
package dev.retrotv.crypt.exception;

public class WrongPaddingException extends RuntimeException {
    public WrongPaddingException() {
        super("해당 알고리즘에서 지원하지 않는 패딩 기법 입니다.");
    }

    public WrongPaddingException(String message) {
        super(message);
    }

    public WrongPaddingException(String message, Throwable throwable) {
        super(message, throwable);
    }
}

File: src/main/kotlin/dev/retrotv/crypto/exception/WrongPaddingException.kt
package dev.retrotv.crypto.exception

/**
 * 잘못된 형식의 패딩 사용 시, 발생할 수 있는 예외를 처리하기 위한 예외입니다.
 *
 * @author  yjj8353
 * @since   1.0.0
 */
class WrongPaddingException : RuntimeException {
    constructor() : super("해당 알고리즘에서 지원하지 않는 패딩 기법 입니다.")
    constructor(message: String) : super(message)
    constructor(message: String, throwable: Throwable) : super(message, throwable)
}
